include/envoy/secret/secret_manager.h

```diff
@@ -21,13 +20,26 @@ class SecretManager {
    * @param secret a protobuf message of envoy::api::v2::auth::Secret.
    * @throw an EnvoyException if the secret is invalid or not supported.
    */
-  virtual void addOrUpdateSecret(const envoy::api::v2::auth::Secret& secret) PURE;
+  virtual void addStaticSecret(const envoy::api::v2::auth::Secret& secret) PURE;
 
   /**
    * @param name a name of the Ssl::TlsCertificateConfig.
    * @return the TlsCertificate secret. Returns nullptr if the secret is not found.
    */
-  virtual const Ssl::TlsCertificateConfig* findTlsCertificate(const std::string& name) const PURE;
+  virtual const Ssl::TlsCertificateConfig*
+  findStaticTlsCertificate(const std::string& name) const PURE;
+
+  /**
+   * Finds and returns a secret provider associated to SDS config. Create a new one
+   * if such provider does not exist.
+   *
+   * @param config_source a protobuf message object contains SDS config source.
+   * @param config_name a name that uniquely refers to the SDS config source
+   * @return the dynamic secret provider.
+   */
+  virtual DynamicSecretProviderSharedPtr
+  findOrCreateDynamicSecretProvider(const envoy::api::v2::core::ConfigSource& config_source,
```

Comment:
per comment above should this be specific to TlsCertificates for now like the static versions?